src/resultSet.js

```diff
@@ -1,37 +1,192 @@
+    COLUMN_UNKNOWN: 0,
+    COLUMN_SCALAR: 1,
+    COLUMN_NODE: 2,
+    COLUMN_RELATION: 3
+}
+
+const ResultSetScalarTypes = {
+    PROPERTY_UNKNOWN: 0,
+    PROPERTY_NULL: 1,
+    PROPERTY_STRING: 2,
+    PROPERTY_INTEGER: 3,
+    PROPERTY_BOOLEAN: 4,
+    PROPERTY_DOUBLE: 5
+}
 
 /**
  * Hold a query result
  */
 class ResultSet {
 	constructor(resp) {
 		this._position = 0;
-		this._statistics = new Statistics(resp[1]);
+        this._resultsCount = 0;
+		this._header = [];
+		this._results = [];
+
+        if(resp.length == 1) {
+            this._statistics = new Statistics(resp[0]);
+        } else{
+    		this.parseResults(resp);
+    		this._resultsCount = this._results.length;
+            this._statistics = new Statistics(resp[3]);
+        }
+	}
+
+	parseResults(resp) {
+        this._string_mapping = this.parseStringMapping(resp);
+        this._header = this.parseHeader(resp);
+        
+        // Discard header types.
+        this._typelessHeader = new Array(this._header.length);
+        for(var i = 0; i < this._header.length; i++) {
+            this._typelessHeader[i] = this._header[i][1];
+        }
 
-		let result = resp[0];
+        this.parseRecords(resp);
+	}
 
-		// Empty result set
-		if (result === null || result.length === 0) {
-			this._header = [];
-			this._totalResults = 0;
-			this._results = [];
-		} else {
-			this._header = result[0];
-			this._totalResults = result.length - 1;
-			this._results = new Array(this._totalResults);
-			for (let i = 0; i < this._totalResults; ++i) {
-				this._results[i] = new Record(this._header, result[i + 1]);
-			}
-		}
+    parseStringMapping(raw_result_set) {
+        // An array of strings, which are referred to
+        // by other parts of the result-set.
+        let string_mapping = raw_result_set[0];
+        return string_mapping;
 	}
 
+	parseHeader(raw_result_set) {
+        // An array of column name/column type pairs.
+        let header = raw_result_set[1];
+        return header;
+	}
+
+	parseRecords(raw_result_set) {
+        let result_set = raw_result_set[2];
+        let records = new Array(result_set.length);
+        this._results = new Array(result_set.length);
+
+        for(var i = 0; i < result_set.length; i++) {
+        	let row = result_set[i];
+            let record = new Array(row.length);
+            for(var j = 0; j < row.length; j++) {
+            	let cell = row[j];
+                if (this._header[j][0] == ResultSetColumnTypes.COLUMN_SCALAR) {
+                    record[j] = this.parseScalar(cell);
+                }
+                else if (this._header[j][0] == ResultSetColumnTypes.COLUMN_NODE) {
+                    record[j] = this.parseNode(cell);
+                }
+                else if (this._header[j][0] == ResultSetColumnTypes.COLUMN_RELATION) {
+                    record[j] = this.parseEdge(cell);
+                }
+                else {
+                    console.log("Unknown column type.\n");
```

Comment:
continue?

Reply:
maybe abort, not sure...

Reply:
Only if you throws an Error in this case